packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -288,7 +288,7 @@ class _GameWidgetState<T extends Game> extends State<GameWidget<T>> {
     if (game is KeyboardEvents) {
       return game.onKeyEvent(event, RawKeyboard.instance.keysPressed);
     }
```

Comment:
```suggestion
    if(!_focusNode.hasPrimaryFocus) {
      return  KeyEventResult.ignored;
    }
    if (game is KeyboardEvents) {
      return game.onKeyEvent(event, RawKeyboard.instance.keysPressed);
    }
```

And set as handled in the end.